cmd/kube-proxy/app/server_others.go

```diff
@@ -157,22 +155,20 @@ func newProxyServer(
 	var ipt [2]utiliptables.Interface
 	dualStack := true // While we assume that node supports, we do further checks below
 
-	if proxyMode != proxyconfigapi.ProxyModeUserspace {
-		// Create iptables handlers for both families, one is already created
-		// Always ordered as IPv4, IPv6
-		if primaryProtocol == utiliptables.ProtocolIPv4 {
-			ipt[0] = iptInterface
-			ipt[1] = utiliptables.New(execer, utiliptables.ProtocolIPv6)
-		} else {
-			ipt[0] = utiliptables.New(execer, utiliptables.ProtocolIPv4)
-			ipt[1] = iptInterface
-		}
+	// Create iptables handlers for both families, one is already created
+	// Always ordered as IPv4, IPv6
+	if primaryProtocol == utiliptables.ProtocolIPv4 {
+		ipt[0] = iptInterface
+		ipt[1] = utiliptables.New(execer, utiliptables.ProtocolIPv6)
+	} else {
+		ipt[0] = utiliptables.New(execer, utiliptables.ProtocolIPv4)
+		ipt[1] = iptInterface
+	}
 
-		for _, perFamilyIpt := range ipt {
-			if !perFamilyIpt.Present() {
-				klog.V(0).InfoS("kube-proxy running in single-stack mode, this ipFamily is not supported", "ipFamily", perFamilyIpt.Protocol())
-				dualStack = false
-			}
+	for _, perFamilyIpt := range ipt {
+		if !perFamilyIpt.Present() {
+			klog.V(0).InfoS("kube-proxy running in single-stack mode, this ipFamily is not supported", "ipFamily", perFamilyIpt.Protocol())
+			dualStack = false
```

Comment:
(lol, this doesn't actually check that `primaryProtocol` is the one that's supported...)